### DISCUSSION TITLE
DirectInput.Effect - IDirectInputEffect

### DISCUSSION THREAD

**Discussion by gagagu** [amerkoleci/Vortice.Windows#473] [General]
Hi, thx for your project!
I am working on a small Force Feedback Joystick Project (see https://github.com/MadCatX/FFBInspector) to translate from the old SlimDX to Vortice.Windows and i have a small problem with the IDirectInputEffect class.

The original code from SlimDX is 

    SlimDX.DirectInput.Effect ef;
    try
     {
        ef = new Effect(ffbDevice, data.effect.effectGuid, eParams);
     }

I am lookin for the corresponding code in Vortice and IDirectInputEffect is not working in the same way and my directx c# isn't good enough to understand how it's working. Can you help me out?

Here are the actual definition for the variables in Vortice

private IDirectInputDevice8 ffbDevice;
DataClasses.FFBEffectData data
EffectParameters eParams = new EffectParameters();

If the conversion works i will upload it to my GitHub Repo.

Thx forward.

**Comment by gagagu** [comment]
I guess this seems to be the solution (untested):

 ef = ffbDevice.CreateEffect(data.effect.effectGuid, eParams);

**Reply by amerkoleci** [reply]
> I guess this seems to be the solution (untested):
> 
> ef = ffbDevice.CreateEffect(data.effect.effectGuid, eParams);

Yes!
